packages/query/src/utils/parse-template.ts

```diff
@@ -29,6 +29,7 @@ export function parseQueryResult(
 	scripts: EnqueuedScriptProps[] | undefined;
 	scriptModules: ScriptModuleProps[] | undefined;
 	bodyClasses: string[] | undefined;
+	is404: boolean;
```

Comment:
@ayushnirwal taking into consideration future-compat with other protentional route-based conditionals: is this the ideal schema shape for consumption?